src/jit/lclvars.cpp

```diff
@@ -236,6 +243,40 @@ void Compiler::lvaInitTypeRef()
     lvaInitArgs(&varDscInfo);
 
     //-------------------------------------------------------------------------
+    // Calculate the argument register usage.
+    //
+    // This will later be used for fastTailCall determination
+    //-------------------------------------------------------------------------
+
+    unsigned   floatingRegCount = 0;
+    unsigned   stackArgCount    = 0;
+    unsigned   stackSize        = 0;
+
+    auto incrementRegCount = [&floatingRegCount, &argRegCount](LclVarDsc* varDsc)
+    {
+        varDsc->IsFloatRegType() ? ++floatingRegCount : ++argRegCount;
+    };
+
+    for (LclVarDsc* varDsc, unsigned argNum = 0; argNum < info.compMethodInfo->args.numArgs; argNum++, varDsc++)
+    {
+        if (varDsc->lvRegNum != REG_STK)
+        {
+            incrementRegCount(varDsc);
+
+#ifdef (FEATURE_MULTIREG_ARGS && !WINDOWS_AMD64_ABI)
+            if (varDsc->lvOtherArgReg != REG_NA)
+            {
+                incrementRegCount(varDsc);
+            }
+#endif // (FEATURE_MULTIREG_ARGS && !WINDOWS_AMD64_ABI)
+        }
+        else
+        {
+            stackSize += varDsc->lvSize();
+        }
```

Comment:
lvSize() can only be used for structs. I think you need a different calculation here for argument stack size.